ui.frontend/src/view/InstanceManager.js

```diff
@@ -151,15 +151,34 @@ export default class InstanceManager {
     removeInstance(event, payload) {
         this.#dispatchModelEvent(event, "removeInstance", payload);
     }
+    #addRemoveInstance(id, eventName, index, event){
+      const model = this.formContainer.getModel(id);
+      const customEvent = new CustomEvent(eventName);
+      customEvent.payload = model.index + index;
+      model.parent.dispatch(customEvent);
+      event.stopPropagation();
+    }
 
     #registerInstanceHandlers(childView) {
-        //doesn't support repeatable panel inside repeatable panel
-        Utils.registerClickHandler(childView.element.parentElement, Constants.DATA_HOOK_ADD_INSTANCE, (event) => {
-            this.addInstance(event)
-        });
-        Utils.registerClickHandler(childView.element.parentElement, Constants.DATA_HOOK_REMOVE_INSTANCE, (event) => {
-            this.removeInstance(event)
-        });
+        if(typeof this?.repeatableParentView?.getRepeatableRootElement === "function"){
+          const parentElement = this.repeatableParentView.getRepeatableRootElement(childView);
+          Utils.registerClickHandler(parentElement, Constants.DATA_HOOK_ADD_INSTANCE, (event) => {
+              const id = event.currentTarget.getAttribute(Constants.DATA_HOOK_ADD_INSTANCE);
+              if(!id){
+                this.addInstance(event);
+              }else {
+                this.#addRemoveInstance(id, 'addInstance', 1, event);
```

Comment:
instead of adding this function calculate index and directly call #dispatchModelEvent.
Also remove todo from line 120 in function #dispatchModelEvent